lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -297,6 +297,51 @@ class StableHLOToTTIRDotGeneralOpConversionPattern
   }
 };
 
+class StableHLOToTTIRGetDimensionSizeOpConversionPattern
+    : public OpConversionPattern<mlir::stablehlo::GetDimensionSizeOp> {
+
+  using OpConversionPattern<
+      mlir::stablehlo::GetDimensionSizeOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(mlir::stablehlo::GetDimensionSizeOp srcOp,
+                  mlir::stablehlo::GetDimensionSizeOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    LogicalResult legalityResult = checkBasicLegality(srcOp, rewriter);
+    if (!legalityResult.succeeded()) {
+      return legalityResult;
+    }
+
+    IntegerType intType = IntegerType::get(getContext(), 32);
+    RankedTensorType outputType = RankedTensorType::get({1}, intType);
+    mlir::OpBuilder builder(getContext());
+    IntegerAttr dimension_attr = builder.getIntegerAttr(
+        intType, static_cast<int32_t>(srcOp.getDimension()));
+
+    rewriter.replaceOpWithNewOp<mlir::tt::ttir::GetDimensionSizeOp>(
+        srcOp, outputType, srcOp.getOperand(), dimension_attr);
+
+    return success();
+  }
+
+private:
+  LogicalResult checkBasicLegality(mlir::stablehlo::GetDimensionSizeOp &srcOp,
+                                   ConversionPatternRewriter &rewriter) const {
+    if (srcOp.getOperand().getType().getShape().empty() &&
+        !srcOp.getOperand().getType().getElementType().isIntOrFloat()) {
+      return rewriter.notifyMatchFailure(srcOp, "Unsupported element type.");
+    }
+    if (!mlir::cast<RankedTensorType>(srcOp.getOperand().getType())) {
+      return rewriter.notifyMatchFailure(srcOp,
+                                         "Unsupported first operator type.");
+    }
+
+    return success();
+  }
```

Comment:
Hmm, I wonder if this is the best way to do this... @ajakovljevicTT have you investigated if any other repo has these situations where there's a need to validate types during conversion? I think implementing this in the verifier of the TTIR op might make more sense, but open to being persuaded :)

Reply:
Added a verify method in the `ttir::GetDimensionSizeOp` instead, should be ok now, please check whan you have the time.